lib/rack/request-id-passthrough.rb

```diff
@@ -32,12 +32,25 @@ def call(env)
 
     def determine_request_id(env)
       request_id = SecureRandom.uuid
+      matches = {}
+      env.each do |key,value|
+        @headers.reverse_each do |header_name|
+          matches[header_name] = value if same_header?(header_name,key)
+        end
+      end
       @headers.reverse_each do |header_name|
-        request_id = env[header_name] if env[header_name]
+        request_id = matches[header_name] if matches[header_name]
       end
+
       request_id
     end
 
+    def same_header?(header_name,key)
```

Comment:
Suggest renaming `key` to something like `env_key`.  I got confused for a minute when looking at the code until I looked back at `determine_request`.